### PR TITLE
Use mysql full text search for search endpoint

### LINKED ISSUE #51: Simplify search

### PR DESCRIPTION
**Pull request by henryc132**
Fixes #51 

@ClaytonPassmore 

Changed search procedure to take advantage of mysql's full text searching.

Decided to leave the Tags table in, even after adding the tags_delimited column since a delimited column is useless for any queries that isn't full text searching

### CODE REVIEW

**Comment by ClaytonPassmore** [comment]
@henryc132 are we going to have any other queries on tags that are going to require that tags table? At this point we aren't doing any categorization or anything like that.

File: Server/App/api/discover.py [hunk 1]
```diff
@@ -42,15 +42,12 @@ def get(self, page, connector):
...
+    @optional_query_string_params('page')
+    def get(self, page, query, connector):
         """Retrieve metadata for decks matching the search string."""
 
-        tags = json.loads(tags_json)
-
-        tags_string = ",".join(tags)
-
-        decks = connector.call_procedure('SEARCH_DECKS', query, tags_string, len(tags), page)
+        decks = connector.call_procedure('SEARCH_DECKS', query, page)
```
Comment (ClaytonPassmore) [inline Server/App/api/discover.py:50]: For all of these `page` optional params, we're going to have to do a `None` check. If it's `None`, we're going to have to set it to 0.

**Review by ClaytonPassmore — CHANGES_REQUESTED** [review]
We should perhaps consider eliminating the `Tags` table.. Let me know your thoughts!

File: Server/App/api/discover.py (hunk 1 above)
Comment (henryc132) [reply Server/App/api/discover.py:50]: I actually had that check in the stored procedure

Comment (ClaytonPassmore) [reply Server/App/api/discover.py:50]: Yup I totally missed that, nice!

**Review by ClaytonPassmore — APPROVED** [review]

**Comment by ClaytonPassmore** [comment]
@henry we can always remove the tags table later. :ship:

**Comment by henryc132** [comment]
I mainly thought of keeping tags in case we want to use it in the future for some sort of advance search or similar feature.

**Comment by ClaytonPassmore** [comment]
@henryc132 I feel like we probably will have other priorities, but whatever, I'm not opposed to keeping it in for now just in case.

**Comment by ClaytonPassmore** [comment]
GTM :motor_boat:

**Pull request merged**